### ISSUE TITLE
Test flake: TestTailFuzz

### ISSUE THREAD

**Issue by jmacd** [lightstep/opentelemetry-prometheus-sidecar#20]
This should be investigated. [Seen in the CircleCI test].(https://app.circleci.com/pipelines/github/lightstep/opentelemetry-prometheus-sidecar/17/workflows/84266bd8-8734-4531-ab41-882efcff36b4/jobs/18)

```
--- FAIL: TestTailFuzz (2.57s)
    tail_test.go:87: corruption after 69042176 bytes: unexpected non-zero byte in padded page
```

**Comment by DanTulovsky** [comment]
I am getting this error in my sidecar on kubernetes right now:

```
level=info ts=2021-02-05T20:03:31.592Z caller=manager.go:104 component=prom_wal msg="Starting Prometheus reader..."
level=info ts=2021-02-05T20:03:31.592Z caller=main.go:272 msg="OpenTelemetry client started"
level=info ts=2021-02-05T20:03:34.399Z caller=supervisor.go:388 supervisor=true msg="sidecar is ready"
level=info ts=2021-02-05T20:03:42.809Z caller=manager.go:166 component=prom_wal msg="reached first record after start offset" start_offset=55985977895 skipped_records=1541
level=info ts=2021-02-05T20:04:01.594Z caller=queue_manager.go:334 component=queue_manager msg="send queue resharding" from=1 to=2
level=info ts=2021-02-05T20:04:16.594Z caller=queue_manager.go:334 component=queue_manager msg="send queue resharding" from=2 to=3
level=info ts=2021-02-05T20:04:31.593Z caller=queue_manager.go:334 component=queue_manager msg="send queue resharding" from=3 to=5
level=info ts=2021-02-05T20:04:34.498Z caller=supervisor.go:409 supervisor=true msg="sidecar is running" samples.processed{}=493839 queue.outcome{outcome:success}=141207
level=info ts=2021-02-05T20:04:41.403Z caller=manager.go:221 component=prom_wal msg="Done processing WAL."
level=info ts=2021-02-05T20:04:41.404Z caller=main.go:254 msg="Prometheus reader stopped"
level=info ts=2021-02-05T20:04:41.404Z caller=main.go:260 msg="Stopping Prometheus reader..."
level=info ts=2021-02-05T20:04:41.405Z caller=queue_manager.go:218 component=queue_manager msg="Stopping remote storage..."
level=info ts=2021-02-05T20:04:41.405Z caller=queue_manager.go:226 component=queue_manager msg="Remote storage stopped."
level=error ts=2021-02-05T20:04:41.405Z caller=main.go:285 err="corruption after 24969216 bytes: unexpected non-zero byte in padded page"
level=error ts=2021-02-05T20:04:41.614Z caller=supervisor.go:97 supervisor=true msg="sidecar failed" err="exit status 1"
level=info ts=2021-02-05T20:05:11.141Z caller=main.go:375 msg="Starting OpenTelemetry Prometheus sidecar" version="(version=0.13.2, branch=main, revision=b02d7fa73811b8dcce34feef0119433e1389424b)" build_context="(go=go1.15.6, user=jmacd@jmacd-lightstep.local, date=20210204-12:03:20)" host_details="(Linux 4.19.0-11-amd64 #1 SMP Debian 4.19.146-1 (2020-09-17) x86_64 prometheus-kube-prometheus-prometheus-0 (none))" fd_limits="(soft=1048576, hard=1048576)"
level=info ts=2021-02-05T20:05:16.103Z caller=supervisor.go:398 supervisor=true msg="sidecar is not ready"
```

And it just keeps going like this...

**Comment by DanTulovsky** [comment]
Leaving the sidecar running and crash looping would eventually lead to grafana not being able to query prometheus, with no obvious errors on the prometheus side.

I disabled the sidecar and let things run normally, all was well.

I have just re-enabled the sidecar and it seems to be populating data again.  Sorry, this isn't very helpful.

**Comment by jmacd** [comment]
I noticed this bug was also filed in the upstream repository:
https://github.com/Stackdriver/stackdriver-prometheus-sidecar/issues/67

It's **great** to see that this is an actual problem affecting the sidecar (#108), not a test-only problem.

**Comment by jmacd** [comment]
This has been explained, see #108. We'll get this fixed.

**Comment by jmacd** [comment]
This was fixed in #112, although #108 remains open.
